affiliation/identity.go

```diff
@@ -255,6 +255,80 @@ func (a *Affiliation) GetIdentityByUser(key string, value string) (*AffIdentity,
 
 }
 
+// GetProfileByUsername ...
+func (a *Affiliation) GetProfileByUsername(username string, projectSlug string) (*AffIdentity, error) {
+	if username == "" && projectSlug == "" {
+		nilKeyOrValueErr := "repository: GetProfileByUsername: username or projectSlug is null"
+		log.Println(nilKeyOrValueErr)
+		return nil, fmt.Errorf(nilKeyOrValueErr)
+	}
+
+	token, err := a.auth0Client.ValidateToken(a.Environment)
+	if err != nil {
+		log.Println(err)
+		return nil, err
+	}
+
+	headers := make(map[string]string, 0)
+	headers["Authorization"] = fmt.Sprintf("%s %s", "Bearer", token)
+	endpoint := a.AffBaseURL + "/affiliation/" + projectSlug + "/get_profile_by_username/" + username
```

Comment:
I think you'll have to url escape the slug and username as they can contain special characters

Reply:
Done. Thanks